File: backend/db/schemas.py
from datetime import date
from pydantic import BaseModel, ConfigDict

class CanchaBase(BaseModel):
	nombre: str
	techada: bool

class CanchaCreate(CanchaBase):
	pass

class CanchaSchema(CanchaBase):
	id: int
	model_config = ConfigDict(from_attributes=True)

class ReservaBase(BaseModel):
	id_cancha: int
	dia: date
	hora: int
	duración_minutos: int
	teléfono: str
	nombre_contacto: str

class ReservaCreate(ReservaBase):
	pass

class ReservaSchema(ReservaBase):
	id: int
	model_config = ConfigDict(arbitrary_types_allowed=True, from_attributes=True)

class ReservaCompletaSchema(BaseModel):
	reserva: ReservaSchema
	cancha: CanchaSchema

File: frontend/src/schemas.ts
export type CanchaSchema = {
	id: number,
	nombre: string,
	techada: boolean,
}

export type ReservaSchema = {
	id: number
	id_cancha: number,
	dia: string,
	hora: number,
	duración_minutos: number,
	teléfono: string,
	nombre_contacto: string,
}

export type ReservaCompletaSchema = {
	reserva: ReservaSchema,
	cancha: CanchaSchema,
}

export type HTTPValidationError = {
	detail: Array<{
		type: string,
		loc: [ string | number ],
		msg: string,
		input: string,
	}>,
}

export type ValidationError = {
	loc: Array<string | number>,
	msg: string,
	type: string,
}

export type HTTPException = {
	detail: string,
}
